### DISCUSSION TITLE
Help with increasing performance in variable manipulation

### DISCUSSION THREAD

**Discussion by Psujj** [Yaskawa-Global/YMConnect#19] [Q&A] [answered]
Hello everyone,
Currently I am developing a project with cpp on linux.
I am trying to send constant information to multiple integer variables in my yrc1000 micro and I am seeing that the "refresh rate" is quite slow. My current approach implies opening and closing ports on every loop and I also have tried leaving the communication port open for the loop, but in both cases the delay keeps happening. 

Any advice onto how to increase the performance of the "refresh rate" would be greatly appreciated!

[https://developer.motoman.com/en/YMConnect/VariableInterface](url) My initial approach is based on this example given by Yaskawa

**Comment by ted-miller** [comment, marked as answer]
Hello @Psujj 
YMConnect is not a deterministic real-time interface. There aren't any settings to increase the performance. (Though I do recommend only opening the connection once for the duration of your program.)

For applications that require real-time performance, I recommend the MotoPlus SDK. This allows you to create a custom executable that runs directly on the robot controller's CPU. It can act as a custom server that communicates with your PC application and controls robot variables at the fastest possible rate.

Documentation for the MotoPlus SDK is available at https://www.motoman.com/en-us/service-training/product-documentation. 
Setup/Installation: 169286-1CD
Overview: 178939-1CD
API Reference: 178941-1CD
